src/module/system/text-editor.ts

```diff
@@ -314,11 +319,32 @@ class TextEditorPF2e extends TextEditor {
                 });
             }
 
+            if (item) {
+                const itemTraits = item.system?.traits?.value;
+                params.templateData = JSON.stringify({
+                    flags: {
+                        pf2e: {
+                            origin: {
+                                name: item.name,
+                                slug: item.slug,
+                                actor: item.actor?.uuid,
+                                traits: itemTraits ? Array.from(itemTraits) : [],
+                                type: item.type,
+                                uuid: item.uuid,
+                            },
+                        },
+                    },
+                });
+            }
+
             // Add the html elements used for the inline buttons
             const html = document.createElement("span");
             html.innerHTML = label;
             html.setAttribute("data-pf2-effect-area", params.type);
             html.setAttribute("data-pf2-distance", params.distance);
+            if (params.templateData) {
+                html.setAttribute("data-pf2-template-data", params.templateData);
+            }
```

Comment:
```suggestion
            if (params.templateData) {
                html.dataset.templateData = params.pf2TemplateData;
            }
```

Reply:
Why do this differently as opposed to doing it the way the other properties are done?